src/nyc_trees/apps/survey/management/commands/send_reminder_emails.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+from __future__ import division
+
+import smtplib
+
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+
+from django.utils.timezone import now
+
+from apps.core.models import TaskRun
+from apps.survey.models import BlockfaceReservation
+from apps.mail.views import send_reservation_reminder
+
+_COMMAND_NAME = 'apps.survey.management.commands.send_reminder_emails'
+_ALREADY_SENT_MESSAGE = ('WARNING: Reservation emails sent already today. '
+                         'Doing nothing!')
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        today = now().date()
+        _, unlocked = TaskRun.objects.get_or_create(name=_COMMAND_NAME,
+                                                    date_started=today)
```

Comment:
> Because we are tracking successful notifications directly on the reservation row, we don't need to enforce that this command runs only once per day, just that it does not run more than once at any given time.

This was my understanding too. It is also possible that I'm not following what `TaskRun` is meant to do correctly, either.

In RRP, I mentioned [`pg_try_advisory_lock`](http://www.postgresql.org/docs/9.3/static/functions-admin.html#FUNCTIONS-ADVISORY-LOCKS-TABLE) as a potential solution to achieving the "it does not run more than once at any given time" objective. It is not tied to a table, and returns `t` immediately if the lock was successfully acquired, `f` otherwise. Closing the connection also releases the lock (good when a failure occurs while you have the lock, but unclear how that works with Django connection pooling). `pg_try_advisory_lock` also has 2 different signatures, but both provide a way to namespace the lock to a specific task type.

This may or may not be useful, but I prototyped a tool that uses the `pg_try_advisory_lock` mechanism after discussing several approaches the JEST has taken in the past to solve this same problem. Happy to discuss that too if anyone wants to go through it.